lib/mongoidable/rspec/controller_matchers.rb

```diff
@@ -0,0 +1,151 @@
+require "rspec/expectations"
+
+module Mongoidable
+  module RSpec
+    module ControllerMatchers
+      extend ::RSpec::Matchers::DSL
+
+      class ControllerHelper < SimpleDelegator
+        def initialize(obj, controller, authorize_pairs)
+          super(obj)
+
+          @controller = controller
+          @authorize_pairs = authorize_pairs
+        end
+
+        def error_message
+          "The #{controller.class.name} did not authenticate the action #{action} against the specified object."
+        end
+
+        def process_action
+          setup_controller
+
+          matchers = authorization_matcher
+          controller._run_process_action_callbacks
+
+          loaded_matcher
+
+          matchers
+        end
```

Comment:
I deleted a previous comment when I did further testing and it failed, and I didn't want to sow too much confusion.

So, I am reproducing it here.

I do not like the [false positives](https://github.com/CardTapp/mortgagemapp/pull/6171/files#diff-8f8416fa87cda8c86da94eb6b8a9c4b1R18) that we're getting in `mortgage_mapp`.  They really drive me nuts.

I want to fix that, and this fixes it for my testing:

```
        def process_action
          setup_controller

          controller._run_process_action_callbacks

          authorization_matcher
        end

        # NOTE:  I prefer the `have_received` vs. the `receive` matchers.  I think that the code is cleaner
        #        for process_action, and here both.  If you really like the other way around, it will work both ways.
        def authorization_matcher
          authorize_pairs.map do |authorize_action, authorized_object|
            matcher = have_received(:authorize!) do |orig_action, orig_object|
              expect(orig_action).to eq authorize_action
              # As silly as it seems to do this, doing this expect will cause an exception when === 
              # returns true but we want it to be false instead.
              expect(orig_object).to eq authorized_object
            end.with(authorize_action, authorized_object)

            matcher.setup_expectation(controller)

            matcher
          end
        end
```

Reply:
Yeah... I have noticed you use have_recieved way too much. Testing in isolation means using receive more than have_received. have_received allows that thing to run which is more often than not not what you want. You want to make sure the resque gets a call to enqueue something not let it enqueue something. 

That said, we will have to discuss this change because I am not following.

Reply:
> You want to make sure the resque gets a call to enqueue something not let it enqueued something.

That's not necessarily how `have_received` works.